collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2519,116 +2552,132 @@ public void loadingComplete(FormLoaderTask task, FormDef formDef) {
 
         final FormController formController = task.getFormController();
         if (formController != null) {
-            int requestCode = task.getRequestCode(); // these are bogus if
-            // pendingActivityResult is
-            // false
-            int resultCode = task.getResultCode();
-            Intent intent = task.getIntent();
+            if (readPhoneStatePermissionRequestNeeded) {
+                requestReadPhoneStatePermission(this, new PermissionListener() {
+                    @Override
+                    public void granted() {
+                        readPhoneStatePermissionRequestNeeded = false;
+                        Collect.getInstance().initProperties();
+                        loadForm();
+                    }
 
-            formLoaderTask.setFormLoaderListener(null);
-            FormLoaderTask t = formLoaderTask;
-            formLoaderTask = null;
-            t.cancel(true);
-            t.destroy();
-            Collect.getInstance().setFormController(formController);
-            supportInvalidateOptionsMenu();
+                    @Override
+                    public void denied() {
+                        finish();
+                    }
+                }, true);
+            } else {
+                int requestCode = task.getRequestCode(); // these are bogus if
```

Comment:
While you're moving this around, I'd recommend doing just a tiny bit of cleanup. At least it would be good to put the comment on one line but really these three variables can just be local to the `if (pendingActivityResult) {` block and then the weird comment isn't needed anymore.

Like you said, it's the (boy|girl)scout rule! 😊

Reply:
Done.